configuration/azure/appconfig/appconfig.go

```diff
@@ -160,32 +160,71 @@ func parseMetadata(meta configuration.Metadata) (metadata, error) {
 func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
 	keys := req.Keys
 	items := make([]*configuration.Item, 0, len(keys))
-	for _, key := range keys {
-		resp, err := r.client.GetSetting(
-			ctx,
-			key,
-			&azappconfig.GetSettingOptions{
-				Label: to.Ptr("label"),
-			})
-		if err != nil {
+
+	if len(keys) == 0 {
+		var err error
+		if items, err = r.getAll(ctx); err != nil {
 			return nil, err
 		}
-
-		item := &configuration.Item{
-			Metadata: map[string]string{},
+	} else {
+		for _, key := range keys {
+			resp, err := r.client.GetSetting(
+				ctx,
+				key,
+				&azappconfig.GetSettingOptions{
+					Label: to.Ptr("label"),
+				})
+			if err != nil {
```

Comment:
If a single key doesn't exist, is an error returned? In that case, what would be best:

- Returning an error (making the entire method fail)
- Or simply ignoring that key (return every other value but skipping that one)

?

Reply:
IMHO entire method should fail as **redis** implementation does. Also secret stores bulk method fail when an error is found fetching a key.

Reply:
Ok consistency is good! So right now, if a value is not set, this returns an error?

Reply:
If a value is not set an empty string is returned. If the call fails for some reason then the method returns an error.